js/main.ts

```diff
@@ -1,36 +1,29 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
 
-// eslint-disable-next-line @typescript-eslint/no-triple-slash-reference
-/// <reference path="./plugins.d.ts" />
-
-import "./globals";
-
-import { assert, log } from "./util";
-import * as os from "./os";
-import { args } from "./deno";
-import { setPrepareStackTrace } from "./error_stack";
-import { replLoop } from "./repl";
-import { xevalMain, XevalFunc } from "./xeval";
-import { setVersions } from "./version";
-import { window } from "./window";
-import { setLocation } from "./location";
-
-// builtin modules
-import * as deno from "./deno";
-
-export default function denoMain(
-  preserveDenoNamespace: boolean = true,
-  name?: string
-): void {
+import "./globals.ts";
+
+import { assert, log } from "./util.ts";
+import * as os from "./os.ts";
+import { args } from "./deno.ts";
+import { setPrepareStackTrace } from "./error_stack.ts";
+import { replLoop } from "./repl.ts";
+import { xevalMain, XevalFunc } from "./xeval.ts";
+import { setVersions } from "./version.ts";
+import { window } from "./window.ts";
+import { setLocation } from "./location.ts";
+import * as Deno from "./deno.ts";
+
+function denoMain(preserveDenoNamespace: boolean = true, name?: string): void {
   const s = os.start(preserveDenoNamespace, name);
 
   setVersions(s.denoVersion, s.v8Version);
 
   // handle `--version`
   if (s.versionFlag) {
-    console.log("deno:", deno.version.deno);
-    console.log("v8:", deno.version.v8);
-    console.log("typescript:", deno.version.typescript);
+    const { console } = window;
```

Comment:
Unnecessary?

Reply:
Necessary - here and elsewhere. Somehow the ambient types changed - I think maybe because we no longer have `libs: ["dom"]` in the tsconfig... I'm not sure.